vmware-tanzu/tanzu-framework
# Extend AntreaConfig to Support Antrea-NSX Registration Workflow

### What this PR does / why we need it
This patch extends antreaConfig to support antreaNsx, once antreaNsx is enabled in antreaConfig, addon controller will
1. create nsxServiceAccount
2. create providerServiceAccount
3. update values.yaml in xxxx-antrea-package secret  with antreaNsx enabled

upon the workload cluster is deleted, addon controller will
1. delete nsxServiceAccount
2. delete providerServiceAccount
3. delete xxxx-antrea-package secret

### Which issue(s) this PR fixes


Fixes #

### Describe testing done for PR
1. create a cluster in namespace antrea-test
2. edit antreaConfig in antrea-test, add 
```
  antreaNsx:
    enable: true
```
3. check if nsxServiceAccount and providerServiceAccount are created in antrea-test
4. check if there is antreaNsx enabled in secret whose name contains antrea-package in its data, use based64 -d to decode the content
5. delete the cluster, check if nsxServiceAccount and providerServiceAccount are deleted
6. check if secret in step 4 is deleted
```
  antreaNsx:
    enable: true
```


### Release note

```release-note

```



### Additional information

#### Special notes for your reviewer
